dd-java-agent/agent-logging/src/main/java/datadog/trace/logging/PrintStreamWrapper.java

```diff
@@ -0,0 +1,67 @@
+package datadog.trace.logging;
+
+import datadog.trace.api.Platform;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.nio.file.Path;
+
+public class PrintStreamWrapper extends PrintStream {
+  private static final int lineSeparatorLength = Platform.isWindows() ? 2 : 1;
+  private boolean captureOutput = false;
```

Comment:
```suggestion
  private volatile boolean captureOutput = false;
```
If we make this `volatile` then any changes before `captureOutput = true` would be visible to other threads checking `if (captureOutput) {`